### DISCUSSION TITLE
get data from collection for embed hook

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#18036] [Archive]
Hi!

I would like to get some CSS from a collection for an embed hook:

```node
module.exports = ({ embed }, { exceptions, services, env }) => {

  const { ItemsService } = services;
  
   /* obviously this doesn't work here: */
  const itemService = new ItemsService("settings", {
    schema: req.schema,
    accountability: req.accountability,
  });
  var items = await itemService.readByQuery({
    fields: [
      "*",
    ],
    filter: {
      page: env.page
    },
  });

  embed(
    'head',
    () => '<style>' + items[0].css + '</style>'
  );
  
};
```

How can I access data from a collection to use for an embed hook?

Kamil

**Comment by rijkvanzanten** [comment]
The embedhandler callback function is currently a regular function (not async), so there's no way to run async operations in that embedder hook 🤔 Mind opening [a new feature request](https://github.com/directus/directus/discussions/new?category=feature-requests)? I think that's a relatively small tweak 🙂

**Reply by br41nslug** [reply]
We can make the `embed` function async however that may give people the false impression that this is a "dynamic value" that can be changed per request while in reality `embed`s are run once at startup, cached and used as static value for the duration of the instance.

**Reply by rijkvanzanten** [reply]
That's fair. Similar problem exists with the `init` hook I suppose 🤔
